omod-2.0/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_0/PatientAllergyController2_0Test.java

```diff
@@ -0,0 +1,157 @@
+import org.openmrs.Allergen;
+import org.openmrs.AllergenType;
+import org.openmrs.Allergies;
+import org.openmrs.Concept;
+import org.openmrs.Patient;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.webservices.rest.web.RestTestConstants2_0;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.Ignore;
+import org.openmrs.module.webservices.rest.SimpleObject;
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.web.context.request.WebRequest;
+import org.openmrs.module.webservices.rest.web.v1_0.controller.MainResourceControllerTest;
+import org.openmrs.module.webservices.rest.web.response.ObjectNotFoundException;
+
+public class PatientAllergyController2_0Test extends MainResourceControllerTest {
+	
+	@Before
+	public void init() throws Exception {
+		executeDataSet(RestTestConstants2_0.ALLERGY_TEST_DATA_XML);
+		executeDataSet(RestTestConstants2_0.OTHER_NON_CODED_CONCEPT_TEST_DATA_XML);
+	}
+	
+	@Override
+	public String getURI() {
+		return "patient/" + RestTestConstants2_0.PATIENT_UUID + "/allergy";
+	}
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.v1_0.controller.MainResourceControllerTest#getUuid()
+	 */
+	@Override
+	public String getUuid() {
+		return RestTestConstants2_0.ALLERGY_UUID; // allergy does not support uuid
+	}
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.v1_0.controller.MainResourceControllerTest#getAllCount()
+	 */
+	@Override
+	public long getAllCount() {
+		return 4;
+	}
+	
+	@Test
+	public void shouldDeleteUniqueAllergy() throws Exception
+	{
+		Allergy allergy = Context.getPatientService().getAllergyByUuid(getUuid());
+		Assert.assertFalse(allergy.isVoided());
+		
+		handle(newDeleteRequest(getURI() + "/" + getUuid(), 
+				new Parameter("reason", "unit test")));
+		allergy = Context.getPatientService().getAllergyByUuid(getUuid());
+		Assert.assertTrue(allergy.isVoided());
+		Assert.assertEquals("unit test", allergy.getVoidReason());
+	}
+
+	@Test
+	public void shouldDeleteAllAllergies() throws Exception
+	{
+		handle(newDeleteRequest(getURI(), 
+				new Parameter("reason", "unit test")));
+		Allergy allergy = Context.getPatientService().getAllergyByUuid(getUuid());
+		Allergies allergies = Context.getPatientService().getAllergies(allergy.getPatient());
+		Assert.assertTrue(allergy.isVoided());
+		Assert.assertEquals("unit test", allergy.getVoidReason());
+		Assert.assertEquals(Allergies.UNKNOWN, allergies.getAllergyStatus());
+	}
+	
+	@Test
+	public void shouldSaveAllergyWithCodedAllergen() throws Exception
+	{
+		Allergy allergy = Context.getPatientService().getAllergyByUuid(getUuid());
+		Patient patient = allergy.getPatient();
+
+		String json = "{\"allergen\": {\"allergenType\":\"DRUG\", \"codedAllergen\": { \"uuid\": \"35d3346a-6769-4d52-823f-b4b234bac3e3\"} }}";
+		handle(newPostRequest(getURI(), json));
+		
+		Allergies allergies = Context.getPatientService().getAllergies(patient);

```

Comment:
By just looking at this test, how do i tell that the allergy was successfully saved?